osu.Game/Overlays/Settings/SettingsItem.cs

```diff
@@ -100,9 +100,23 @@ public virtual Bindable<T> Current
 
         public IEnumerable<string> Keywords { get; set; }
 
-        public override bool IsPresent => base.IsPresent && MatchingFilter;
+        private bool matchingFilter;
+
+        public bool MatchingFilter
+        {
+            get => matchingFilter;
+            set
+            {
+                bool wasPresent = IsPresent;
 
-        public bool MatchingFilter { get; set; } = true;
+                matchingFilter = value;
+
+                if (IsPresent != wasPresent)
+                    Invalidate(Invalidation.Presence);
```

Comment:
I don't know about this. In an ideal world, this sort of thing should be handled by framework. There are several other `IsPresent` overrides in the game alone even and as far as I understand every single one is susceptible to this same bug of invalidations not firing on custom presence criteria changing. And not every one of those will be so easily fixable as this one (the ones that check `Scheduler.HasPendingTasks` certainly aren't going to be...)

Reply:
For sure, but I'm not really sure how it should be approached framework-side. If anything, we probably don't want to allow `IsPresent` to be overridable in the first place, to avoid this weirdness happening.

If I were to *guess* of a proper fix, it might be to have something similar to the `ShouldBeAlive` and `IsAlive` flow (in which the former is an overridable property, while the latter is updated internally to the state of the former).

But for now, every override of `IsPresent` must invalidate `Presence`, and I'm unsure how that should be achieved for all overrides, so I sticked into fixing `SettingsOverlay` which is the most important here.

Reply:
Is this the kind of case we want to use `AddLayout` instead?

Reply:
`AddLayout` solves the inverse problem doesn't it? `LayoutMember` is for when you want to recompute something when an invalidation happens. But the issue here is that due to `IsPresent` being a computed read-only property, framework can't know when it changes to fire off an invalidation without adding some flow that would check whether the value changed between consecutive reads.

Reply:
Ah yep, that makes sense.

I guess this is okay for now?